### DISCUSSION TITLE
prerendered error

### DISCUSSION THREAD

**Discussion by sundarsk123** [vercel/next.js#68024] [Help] [answered]
### Summary

![Screenshot (392)](https://github.com/user-attachments/assets/03f3be08-340c-4350-b8d2-efa8846e424a)
if i push vercel i face the prerendered error occur how to solve

**Comment by icyJoseph** [comment, marked as answer]
Hi,

This is a common pitfall: https://nextjs-faq.com/fetch-api-in-rsc

Often, and without a good reason, people fetch from their own server, as you render. This won't work during pre-rendering because there is no server to receive the request.

You need to refactor, to get the data you need from your real source, directly on the component, as you render. Server Components, do no send their JavaScript implementation to client, so there is no problem with importing modules, and other stuff in them.

You might be able to reproduce this locally:

- remove `.next` folder
- run your build script, like, `npm run build`
